spec/rspec/mocks/test_double_spec.rb

```diff
@@ -48,6 +48,17 @@ module Mocks
           end
         end
       end
+
+      pending "test doubles able to register themselves with ObjectInspector" do
+        allow_any_instance_of(TestDouble).to receive(:name).and_return("FooBar")
+        double('User')
+
+        RSpec::Support::ObjectInspector.register TestDouble do |user|
+          "User (#{user.name})"
+        end
+
+        expect(double.inspect).to eq("User (FooBar)")
```

Comment:
> That's not right.  We're not touching the `double.inspect` method.  We're registering the type with `ObjectInspector` so that `ObjectInspector.inspect(double)` can be used to format the double in failure messages.  The spec should likewise use `ObjectInspector.inspect`.

Ah yeah you're right, but still the point was that we shouldn't be registering the double in the spec, and we shouldn't be stubbing name on the double.